### ISSUE TITLE
Support interpolation in mongo processor

### ISSUE THREAD

**Issue by BrenQ** [Jeffail/benthos#982]
Hi, I was wondering if it is possible to support [interpolations](https://www.benthos.dev/docs/configuration/interpolation/) in the mongo processor. For example, I wanted to be able to do something like this:

Depending on the country, the name of the collection changes.

```
- mongodb:
         url: ${DB_MONGO_URL}
         database: ${DB_MONGO_NAME}
         collection: '${! meta("country") }-property'
         username: ${DB_USER}
         password: ${DB_PASS}
         operation: find-one
         filter_map: |-
            root.id = this.id
            hint_map: ""
         write_concern:
            w: ""
            j: false
            w_timeout: "60s"
```

Error: 

`{"@timestamp":"2021-11-30T17:13:47-03:00","@service":"benthos","component":"benthos.pipeline.processor.2.0","json":"{\"error\":\"(InvalidNamespace) Invalid collection name specified 'ads.${! meta(\\\"country\\\")}-property\",\"is_error\":true}","meta":"{\"action\":\"delete\",\"country\":\"co\",\"entity\":\"property\",\"gcp_pubsub_publish_time_unix\":\"1638303226\",\"key\":\"1458513\"}","level":"DEBUG","message":" Avro data parsed"}`

Benthos version

```
Version: 3.59.0
Date: 2021-11-26T04:48:24Z
```

Thank you for everything. It's a great project ❤️

**Comment by mihaitodor** [comment]
Hey @BrenQ I pushed a fix for this in https://github.com/Jeffail/benthos/pull/1054.

**Comment by BrenQ** [comment]
> Hey @BrenQ I pushed a fix for this in #1054.

Hey @mihaitodor , I've seen it and it looks very good. It can be very useful to reduce code and resources. Thank you very much for taking it into account !
